jujugui/static/gui/src/app/components/user-profile/model-list/test-model-list.js

```diff
@@ -241,12 +252,114 @@ describe('UserProfileModelList', () => {
+    output = component.getRenderOutput();
+    const expected = (
+      <juju.components.ConfirmationPopup
+        buttons={output.props.children[2].props.buttons}
+        message={'Are you sure you want to destroy sandbox? All the ' +
+          'applications and units included in the model will be destroyed. ' +
+          'This action cannot be undone.'}
+        title="Destroy model" />);
+    assert.deepEqual(output.props.children[2], expected);
+  });
+
+  it('can destroy a model', () => {
+    const addNotification = sinon.stub();
+    const uuids = {};
+    models.forEach(model => {
+      uuids[model.uuid] = null;
+    });
+    const destroyModels = sinon.stub().callsArgWith(1, null, uuids);
+    const listModelsWithInfo = sinon.stub().callsArgWith(0, null, models);
+    const component = jsTestUtils.shallowRender(
+      <juju.components.UserProfileModelList
+        addNotification={addNotification}
+        currentModel={'model1'}
+        destroyModels={destroyModels}
+        listModelsWithInfo={listModelsWithInfo}
+        switchModel={sinon.stub()}
+        user={user} />, true);
+    const instance = component.getMountedInstance();
+    instance._displayConfirmation({name: 'spinach/my-model'});
+    const output = component.getRenderOutput();
+    output.props.children[2].props.buttons[1].action();
+    assert.equal(destroyModels.callCount, 1, 'destroyModels not called');
+    output = component.getRenderOutput();
+    // The confirmation should now be hidden.
+    assert.isUndefined(output.props.children[2], 'Confirmation not hidden');
+    assert.equal(addNotification.callCount, 1, 'addNotification not called');
+    assert.deepEqual(addNotification.args[0][0], {
+      title: 'Model destroyed',
+      message: 'The model is currently being destroyed.',
+      level: 'important'
+    }, 'Notification message does not match expected.');
+  });
+
+  it('can cancel destroying a model', () => {
+    const destroyModels = sinon.stub();
+    const listModelsWithInfo = sinon.stub().callsArgWith(0, null, models);
+    const component = jsTestUtils.shallowRender(
+      <juju.components.UserProfileModelList
+        addNotification={sinon.stub()}
+        currentModel={'model1'}
+        destroyModels={destroyModels}
+        listModelsWithInfo={listModelsWithInfo}
+        switchModel={sinon.stub()}
+        user={user} />, true);
+    const instance = component.getMountedInstance();
+    instance._displayConfirmation({name: 'spinach/my-model'});
+    const output = component.getRenderOutput();
+    output.props.children[2].props.buttons[0].action();
+    assert.equal(destroyModels.callCount, 0, 'destroyModels was called');
+    output = component.getRenderOutput();
+    // The confirmation should now be hidden.
+    assert.isUndefined(output.props.children[2], 'Confirmation not hidden');
+  });
+
+  it('can display an error if the model fails to destroy', () => {
```

Comment:
Nice tests! I think a couple of error cases are missing:
- trying to destroy a controller;
- receiving a specific model error, not a global one.